libp2p/protocols/kademlia/kademlia.nim

```diff
@@ -68,6 +74,46 @@ proc waitRepliesOrTimeouts(
 
   return (receivedReplies, failedPeers)
 
+proc dispatchPutVal(
+    kad: KadDHT, peer: PeerId, key: Key, val: seq[byte]
+): Future[void] {.
+    async: (raises: [CancelledError, DialFailedError, ValueError, LPStreamError])
+.} =
+  info "dialing", me = kad.rtable.selfId, peer = peer, key = key, val = val[0 .. 3]
+  let conn = await kad.switch.dial(peer, KadCodec)
+
+  defer:
+    await conn.close()
+
+  let msg = Message(
+    msgType: MessageType.putValue,
+    record: some(Record(key: some(key.getBytes()), value: some(val))),
+  )
+
+  info "writing"
```

Comment:
This should be a trace log

Reply:
Left over from dev. I think it can go, or would you prefer it to stay, with more info detail, as a trace?

Reply:
[08c994b](https://github.com/vacp2p/nim-libp2p/pull/1582/commits/08c994b32af9064ffdcdb19fff8c0fbce7dc8ed7)